x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/helpers.tsx

```diff
@@ -115,3 +118,17 @@ export function getDataSourceProps(dataSource: DiffableAllFields['data_source'])
 
   return assertUnreachable(dataSource);
 }
+
+export const getFormattedSectionLabel = (
+  label: string,
+  field: string,
+  showModifiedFields: boolean
+) =>
+  showModifiedFields ? (
+    <EuiFlexGroup gutterSize="xs" alignItems="center">
+      {label}
+      <CustomizedPrebuiltRulePerFieldBadge field={field} />
+    </EuiFlexGroup>
+  ) : (
+    label
+  );
```

Comment:
I've already left a comment regarding that above. Let's simplify this by getting rid of the condition, extracting the layout to `RuleFieldName` and letting the context decide whether the "Modified" labels should be shown.

Reply:
These field list components are used elsewhere in the code (both rule upgrade flyouts and alert expanded flyouts) that aren't under the same context umbrella which is why I have this distinguisher here. We could move this into the component itself but I believe the `showModifiedFields` param is still needed unless we're content to add the context elsewhere in the code which I'd argue isn't the best path forward.